File: src/HolzShots/Input/Actions/FullscreenCommand.vb
Imports HolzShots.Composition.Command
Imports HolzShots.Drawing
Imports HolzShots.Threading

Namespace Input.Actions
    <Command("captureEntireScreen")>
    Public Class FullscreenCommand
        Inherits CapturingCommand

        Public Overrides Async Function Invoke(parameters As IReadOnlyDictionary(Of String, String), settingsContext As HSSettings) As Task
            ' TODO: Add proper assertion
            ' Debug.Assert(ManagedSettings.EnableFullscreenScreenshot)

            ' TODO: Re-add proper if condition
            ' If ManagedSettings.EnableFullscreenScreenshot Then

            Dim shot = CaptureFullScreen()
            Debug.Assert(shot IsNot Nothing)
            Await ProcessCapturing(shot, settingsContext).ConfigureAwait(True)
            'End If
        End Function

        Shared Function CaptureFullScreen() As Screenshot
            Using prio As New ProcessPriorityRequest()
                Dim screen = ScreenshotCreator.CaptureScreenshot(SystemInformation.VirtualScreen)
                Return Screenshot.FromImage(screen, Cursor.Position, ScreenshotSource.Fullscreen)
            End Using
        End Function
    End Class
End Namespace

File: src/HolzShots/Input/Action/FullscreenCommand.cs
using System.Diagnostics;
using System.Windows.Forms;
using HolzShots.Composition.Command;
using HolzShots.Drawing;
using HolzShots.Threading;

namespace HolzShots.Input.Actions
{
    [Command("captureEntireScreen")]
    public class FullscreenCommand : CapturingCommand
    {
        protected override async Task InvokeInternal(IReadOnlyDictionary<string, string> parameters, HSSettings settingsContext)
        {
            // TODO: Add proper assertion
            // Debug.Assert(ManagedSettings.EnableFullscreenScreenshot)

            // TODO: Re-add proper if condition
            // If ManagedSettings.EnableFullscreenScreenshot Then

            var shot = CaptureFullScreen();
            Debug.Assert(shot != null);
            await ProcessCapturing(shot, settingsContext).ConfigureAwait(true);
        }

        public static Screenshot CaptureFullScreen()
        {
            using var prio = new ProcessPriorityRequest();
            var screen = ScreenshotCreator.CaptureScreenshot(SystemInformation.VirtualScreen);
            return Screenshot.FromImage(screen, Cursor.Position, ScreenshotSource.Fullscreen);
        }
    }
}
